src/get_model_metadata_impl.cpp

```diff
@@ -58,13 +58,7 @@ Status GetModelMetadataImpl::getModelStatus(
         }
     }
 
-    if (ModelVersionState::AVAILABLE != instance->getStatus().getState()) {
-        return StatusCode::MODEL_MISSING;
```

Comment:
You are changing responses here(from MODEL_MISSING to all possible retcodes from waitForLoaded). I am personally OK with this change since the responses will be more adequate and give more details. However we may sync with validation before to inform about suspected impact.

Reply:
Changes from validation perspective (reported to validation team):

  | Before Status | Before Message | After Status | After Message
-- | -- | -- | -- | --
Model does not exist | NOT_FOUND | Model with requested name is not found | NOT_FOUND | Model with requested name is not found
Model is available | OK | N/A | OK | N/A
Model is loading/errored during load | NOT_FOUND | Model with requested name and/or version is not found | NOT_FOUND | Model with requested version is not loaded yet
Model is retired | NOT_FOUND | Model with requested name and/or version is not found | NOT_FOUND | Model with requested version is retired